extensions/security-webauthn/runtime/src/main/java/io/quarkus/security/webauthn/WebAuthnSecurity.java

```diff
@@ -1,84 +1,418 @@
-        }
+        // apply config defaults
+        this.rpId = config.relyingParty().id().orElse(null);
+        this.rpName = config.relyingParty().name();
+        this.origin = config.origin().orElse(null);
+        this.challengeCookie = config.challengeCookieName();
+        this.challengeUsernameCookie = config.challengeUsernameCookieName();
+        this.challengeLength = config.challengeLength().orElse(64);
+        this.userPresenceRequired = config.userPresenceRequired().orElse(false);
+        this.timeout = config.timeout().orElse(Duration.ofSeconds(60));
         if (config.pubKeyCredParams().isPresent()) {
-            options.setPubKeyCredParams(config.pubKeyCredParams().get());
+            this.pubKeyCredParams = new ArrayList<>(config.pubKeyCredParams().get().size());
+            for (PublicKeyCredential publicKeyCredential : config.pubKeyCredParams().get()) {
+                this.pubKeyCredParams.add(new PublicKeyCredentialParameters(PublicKeyCredentialType.PUBLIC_KEY,
+                        COSEAlgorithmIdentifier.create(publicKeyCredential.coseId())));
+            }
+        } else {
+            this.pubKeyCredParams = new ArrayList<>(2);
+            this.pubKeyCredParams
+                    .add(new PublicKeyCredentialParameters(PublicKeyCredentialType.PUBLIC_KEY, COSEAlgorithmIdentifier.ES256));
+            this.pubKeyCredParams
+                    .add(new PublicKeyCredentialParameters(PublicKeyCredentialType.PUBLIC_KEY, COSEAlgorithmIdentifier.RS256));
         }
-        if (config.requireResidentKey().isPresent()) {
-            options.setRequireResidentKey(config.requireResidentKey().get());
-        }
-        if (config.timeout().isPresent()) {
-            options.setTimeoutInMilliseconds(config.timeout().get().toMillis());
+        this.authenticatorAttachment = config.authenticatorAttachment().orElse(null);
+        this.userVerification = config.userVerification().orElse(UserVerification.DISCOURAGED);
+        this.residentKey = config.residentKey().orElse(ResidentKey.DISCOURAGED);
+        this.attestation = config.attestation().orElse(Attestation.NONE);
+        // create the webauthn4j manager
+        this.webAuthn = makeWebAuthn(vertx, config);
+        this.random = VertxContextPRNG.current(vertx);
+    }
+
+    private String randomBase64URLBuffer() {
+        final byte[] buff = new byte[challengeLength];
+        random.nextBytes(buff);
+        return Base64UrlUtil.encodeToString(buff);
+    }
+
+    private WebAuthnAsyncManager makeWebAuthn(Vertx vertx, WebAuthnRunTimeConfig config) {
+        if (config.attestation().isPresent()
+                && config.attestation().get() == WebAuthnRunTimeConfig.Attestation.ENTERPRISE) {
+            TrustAnchorAsyncRepository something;
+            // FIXME: make config name configurable?
+            Optional<TlsConfiguration> webauthnTlsConfiguration = certificates.get("webauthn");
```

Comment:
It's definitely not related to TLS, but for some reason the Quarkus certificate registry is called  the TLS registry https://quarkus.io/guides/tls-registry-reference
This is just how every certificate list (and trust stores, private, public keys) are configured in Quarkus.
I do agree that the name implies a single purpose that is limiting, @cescoffier

Reply:
Well, @cescoffier himself told me to use this, to centralise everything 🤷

Reply:
@FroMage TLS registry entries have properties related to TLS like handshake timeout, host verification options, reload related one, etc. If the attestation is not happening on the TLS path, it can be confusing for Quarkus WebAuthn users having to refer to TLS registry...
TLS registry is a great concept, but perhaps it can be enhanced over time to extend say `TrustStoreRegistry` instead of having to refer to it on the non-TLS paths